source/patterns/@aws-solutions-constructs/aws-cloudfront-s3/lib/index.ts

```diff
@@ -124,25 +129,68 @@ export class CloudFrontToS3 extends Construct {
       });
       this.s3Bucket = buildS3BucketResponse.bucket;
       this.s3LoggingBucket = buildS3BucketResponse.loggingBucket;
-      bucket = this.s3Bucket;
+      originBucket = this.s3Bucket;
     } else {
-      bucket = props.existingBucketObj;
+      originBucket = props.existingBucketObj;
     }
 
-    this.s3BucketInterface = bucket;
+    this.s3BucketInterface = originBucket;
 
+    // Define the CloudFront Distribution
     const cloudFrontDistributionForS3Response = defaults.CloudFrontDistributionForS3(
       this,
       this.s3BucketInterface,
       props.cloudFrontDistributionProps,
       props.insertHttpSecurityHeaders,
-      props.originPath,
       props.cloudFrontLoggingBucketProps,
       props.responseHeadersPolicyProps
     );
     this.cloudFrontWebDistribution = cloudFrontDistributionForS3Response.distribution;
     this.cloudFrontFunction = cloudFrontDistributionForS3Response.cloudfrontFunction;
     this.cloudFrontLoggingBucket = cloudFrontDistributionForS3Response.loggingBucket;
-  }
+    this.originAccessControl = cloudFrontDistributionForS3Response.originAccessControl;
 
-}
+    // Attach the OriginAccessControl to the CloudFront Distribution, and remove the OriginAccessIdentity
+    const l1CloudFrontDistribution = this.cloudFrontWebDistribution.node.defaultChild as cloudfront.CfnDistribution;
+    l1CloudFrontDistribution.addPropertyOverride('DistributionConfig.Origins.0.OriginAccessControlId', this.originAccessControl?.attrId);
```

Comment:
We discussed this in person already but for transparency on the PR, I removed the line from our demo/sample based on #831 that sets the value of `DistributionConfig.Origins.0.S3OriginConfig.OriginAccessIdentity` to `''`. This is because that solution still creates an OAI and orphans it by setting the reference to an empty string. The solution proposed by this PR creates an entirely new `S3OacOrigin` object that's accepted by the CloudFront distribution as an `origin` value that completely removes the OAI from the picture, which allows us to deploy into new regions.